baybe/surrogates/base.py

```diff
@@ -56,112 +100,261 @@ class Surrogate(ABC, SerialMixin):
 
-        # Evaluate the posterior distribution
-        mean, covar = self._posterior(candidates)
+    def _make_input_scaler(self, searchspace: SearchSpace) -> ColumnTransformer:
+        """Make the input scaler for transforming computational dataframes."""
+        # Create a composite scaler from parameter-wise scaler objects
+        mapping: dict[tuple[int, ...], InputTransform] = {}
+        for p in searchspace.parameters:
+            if (factory := self._make_parameter_scaler_factory(p)) is None:
+                continue
+            idxs = searchspace.get_comp_rep_parameter_indices(p.name)
+            transformer = factory(len(idxs))
+            mapping[idxs] = transformer
+        scaler = ColumnTransformer(mapping)
 
-        # Apply covariance transformation for marginal posterior models
-        if not self.joint_posterior:
-            # Convert to tensor containing covariance matrices
-            covar = torch.diag_embed(covar)
+        # Fit the scaler to the parameter bounds
+        scaler.fit(to_tensor(searchspace.comp_rep_bounds))
 
-        # Add small diagonal variances for numerical stability
-        covar.add_(torch.eye(covar.shape[-1]) * _MIN_VARIANCE)
+        return scaler
 
-        return mean, covar
+    def _make_output_scaler(
+        self, objective: Objective, measurements: pd.DataFrame
+    ) -> OutcomeTransform | _NoTransform:
+        """Make the output scaler for transforming computational dataframes."""
+        if (factory := self._make_target_scaler_factory()) is None:
+            return _IDENTITY_TRANSFORM
 
-    @abstractmethod
-    def _posterior(self, candidates: Tensor) -> tuple[Tensor, Tensor]:
-        """Perform the actual posterior evaluation logic.
+        # TODO: Multi-target extension
+        scaler = factory(1)
+
+        # TODO: Consider taking into account target boundaries when available
+        scaler(to_tensor(objective.transform(measurements)))
+        scaler.eval()
+
+        return scaler
+
+    def posterior(self, candidates: pd.DataFrame, /) -> Posterior:
+        """Compute the posterior for candidates in experimental representation.
+
+        Takes a dataframe of parameter configurations in **experimental representation**
+        and returns the corresponding posterior object. Therefore, the method serves as
+        the user-facing entry point for accessing model predictions.
 
-        In contrast to its public counterpart
-        :func:`baybe.surrogates.Surrogate.posterior`, no data
-        validation/transformation is carried out but only the raw posterior computation
-        is conducted.
+        Args:
+            candidates: A dataframe containing parameter configurations in
+                **experimental representation**.
+
+        Raises:
+            ModelNotTrainedError: When called before the model has been trained.
+
+        Returns:
+            A :class:`botorch.posteriors.Posterior` object representing the posterior
```

Comment:
The problem is that the intersphinx mapping is not set up correctly in `conf.py`, but I don't know how to do it for botorch / if it is even possible. As far as I've seen in the docs, you need to add a link to a location where a `objects.inv` file is available, but I don't know where this would be. Any ideas?

Reply:
Unfortunately only "try and error"/"look how others do it". Can have a look, but since this is really minor, I'd prefer to skip if that is too much work (i.e. more than adjusting this very link here).